stock_reserve_sale/model/sale_order_line.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2013 Camptocamp SA - Guewen Baconnier
+# Copyright 2023 - Hugo Córdoba - FactorLibre - (hugo.cordoba@factorlibre.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+from odoo.exceptions import UserError, except_orm
+from odoo.tools.translate import _
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    def _get_line_rule(self):
+        """Get applicable rule for this product
+        Reproduce get suitable rule from procurement
+        to predict source location
+        """
+        StockRule = self.env["stock.rule"]
+        product = self.product_id
+        product_route_ids = [
+            x.id for x in product.route_ids + product.categ_id.total_route_ids
+        ]
+        rules = StockRule.search(
+            [("route_id", "in", product_route_ids)],
+            order="route_sequence, sequence",
+            limit=1,
+        )
+        if not rules:
+            warehouse = self.order_id.warehouse_id
+            wh_routes = warehouse.route_ids
+            wh_route_ids = [route.id for route in wh_routes]
```

Comment:
We can optimize this
```suggestion
            wh_route_ids = warehouse.route_ids.ids
```